File: Leetcode75/isSubsequence.py
def isSubsequence(s : str , t: str)->bool:

    """
    problem
    Given two strings s and t, return true if s is a subsequence of t, or false otherwise.

    string t remove some char(can be none) -> s

    two poin

    check từng giá trị s trong t
    """
    if len(s)>len(t):
        return False
    if len(s)==0:
        return True
    index_s = 0 
    index_t = 0
    while index_t < len(t):
        if s[index_s] == t[index_t]: 
            index_s+=1
            if index_s >= len(s):
                return True
        index_t+=1
    return False
s = 'abc'
t = 'adsfsc'
print(isSubsequence(s,t))

File: Leetcode75/IsSubsequence.cpp
#include <iostream>
using namespace std;

bool IsSubsequence(string s, string t){

    int index_s = 0;
    int index_t = 0;
    if(t.size()<s.size()) return false;
    // if (s.size()==0) return true;

    while (index_t < t.size() && index_s < s.size())
    {
        if(s[index_s]==t[index_t]){
            index_s += 1;
        }
        
        index_t+=1;
    }
    
    return index_s == s.size();
}



int main(){
    string str1 = "axc" ;
    string str2 = "ahbgdc" ;
    cout<<IsSubsequence(str1,str2);
}